lib/namespacing-transform.js

```diff
@@ -26,6 +44,20 @@ module.exports = class NamespacingTransform {
         if (node.tag.indexOf('::') > -1) {
           node.tag = node.tag.replace('::', '@');
         }
+      },
+      MustacheStatement(node) {
+        if (node.path.original !== 'component') {
+          // we don't care about non-component expressions
+          return;
+        }
+        rewriteOrWrapComponentParam(node, b);
+      },
+      SubExpression(node) {
+        if (node.path.original !== 'component') {
+          // we don't care about non-component expressions
+          return;
+        }
+        rewriteOrWrapComponentParam(node, b);
       }
```

Comment:
Also need to handle `BlockStatement‘ (for `{{#component ‘blah::bar-baz’}}{{/component}}`)

Reply:
Totally forgot this. I've also added tests for the `{{component (some-helper)}}` as I was not covering this. Thanks!